### ISSUE TITLE
Improve DoD performance in Gdocs

### ISSUE THREAD

**Issue by larsyencken** [owid/owid-grapher#3757]
## Problem

We have over 500 DoDs in no particular order in a gigantic Google Doc. We would like to organise them better, e.g. by area and/or topic.

## Proposed solution

Create a new Gdoc type: `dod-definitions` and split the current `fragment` Gdoc to multiple `dod-definitions` documents.

This will give us all the existing and future Gdoc affordances: (e.g content graph reference checks, publishing, versioning) and will give authors the ability to bookmark their particularly salient dictionaries (as opposed to a one-gdoc-per-dod approach, which would likely always require the admin to find) 

**Todos**
- [ ] Create `dod-definitions` gdoc type
  - [ ] Validate that there aren't ID conflicts between different (or within the same) gdocs before you can publish
- [ ] Copy current dod document into it to seamlessly migrate to (authors can then choose to split it up once the PR is merged)
- [ ] Add an admin page to search through all current dods present in all `dod-definitions` gdocs
  - [ ] Add a new admin endpoint to fetch `dods-with-source-gdoc-ids.json`
  - [ ] Filter by tag
  - [ ] Search by id and content 
- [ ] Update dods.json code to union all published `dod-definitions` documents

**Comment by larsyencken** [comment]
We discussed that we shouldn't spend more than (say) 3h on any LLM investment here, otherwise it's better just to take the areas from our nav and sort DoDs into those areas manually as you go.

**Comment by larsyencken** [comment]
Call this "nice to have" since it's probably only a slight time saver, and overall it's less important than getting good usage metrics on DoDs, for example.

**Comment by JoeHasell** [comment]
Fiona reports finding that the Gdoc 'can sometimes be a bit laggy'.

**Comment by spoonerf** [comment]
@paarriagadap mentions in #tech-ideas: "I realized that the [DoD file](https://docs.google.com/document/d/1pwTrbEKOy4D3LPbpbNaivQyYVv2KTePpw-P4F5fMNjA/edit?tab=t.0) works very slowly, given that it currently has 170 (!) pages. It would be nice if, in the future, we could have a tool in admin to process them more efficiently."

**Comment by danyx23** [comment]
Queuing this up for triage again for next week so that we can decide how we want to tackle this in an upcoming batch cycle

**Comment by danyx23** [comment]
We should turn DoD into a new Gdoc document type so that we can use multiple documents instead of one huge one. Breaking the DoDs into topics will be some manual work as well (maybe an llm can help?)

**Comment by danyx23** [comment]
@ikesau given that Fiona recently complained about this, it would be great if it could fit into your batch cycle.

**Comment by ikesau** [comment]
We still think there might be a valid case for the admin-first markdown method of specifying DoDs. I'll ask the authors in slack if they have a preference
